### ISSUE TITLE
argument flag `--cli-version nightly` requires `--cli-scala-version 3.2.0` to be passed explicitly

### ISSUE THREAD

**Issue by scalway** [VirtusLab/scala-cli#1345]
**Version(s)**
```
$ scala-cli about
Scala CLI version: 0.1.12
Scala version (default): 3.1.3
```

**Describe the bug / To Reproduce**
Calling command `scala-cli --cli-version nightly about` copied directly from [documentation](https://scala-cli.virtuslab.org/docs/commands/basics/#scala-cli-version) throws exception.

```
Exception in thread "main" java.lang.NoSuchMethodError: 'long scala.runtime.LazyVals$.getOffsetStatic(java.lang.reflect.Field)'
	at scala.cli.ScalaCli$.<clinit>(ScalaCli.scala:19)
	at scala.cli.ScalaCli.main(ScalaCli.scala)
```
When last snapshot from [here](https://oss.sonatype.org/content/repositories/snapshots/org/virtuslab/scala-cli/core_3/) is specified `scala-cli --cli-version 0.1.12-8-g929dfc35-SNAPSHOT about` it works as expected:

```
Scala CLI version: 0.1.12-8-g929dfc35-SNAPSHOT
Scala version (default): 3.1.3
```

**Comment by Gedochao** [comment]
So the cause for this is our recent bump to `3.2.0` and related incompatibilities. The CLI version isn't detected properly when you try to run the newest snapshots or the nightly version with `--cli-version`. 

This should be fixed after the next release (#1276)

In the meantime, you can pass the `--cli-scala-version` option explicitly, which should fix the problem.
The following workaround should do the trick.
```
▶ scala-cli --cli-version nightly --cli-scala-version 3.2.0  
Welcome to Scala 3.2.0 (17.0.2, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                                             
scala> 
```

**Comment by Gedochao** [comment]
```
▶ scala-cli version
Scala CLI version: 0.1.13
Scala version (default): 3.2.0                                                                             
▶ scala-cli --cli-version nightly
Fetching Scala CLI 0.1.13-2-g9a757cab-SNAPSHOT
Downloading internal stub dependency
Welcome to Scala 3.2.0 (17.0.2, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.                                                                             
scala> 
```

This is fixed for now, just update your `scala-cli` to `0.1.13`.
However, if we don't want to run into this whenever the `minor` version of Scala is updated, we might want to work on #1356 soon.
Closing this ticket in the meantime.
